django/test/testcases.py

```diff
@@ -1030,8 +1030,18 @@ def _fixture_teardown(self):
                          allow_cascade=self.available_apps is not None,
                          inhibit_post_migrate=inhibit_post_migrate)
 
-    def assertQuerysetEqual(self, qs, values, transform=repr, ordered=True, msg=None):
+    def assertQuerysetEqual(self, qs, values, transform=None, ordered=True, msg=None):
+        values = list(values)
+
+        if values:
+            if transform is None:
+                if isinstance(values[0], str):
```

Comment:
If we do it the other way around by using an identity transformer `if isinstance(values[0], qs.model)` we would work around the `QuerySet.values_list('charfield', flat=True)` use case.

Reply:
I hadn't thought of flattened values lists.  Have updated the code for this (I think it's a little bit more complicated than just doing the `if` test the other way round) and I've added a test.